### ISSUE TITLE
Initial transition to substate does not trigger OnTransitioned event

### ISSUE THREAD

**Issue by DeepakParamkusam** [dotnet-state-machine/stateless#413]
Consider:
```
public static void Main(string[] args)
{
    var sm = new StateMachine<State, Trigger>(State.A);
    sm.OnTransitioned(t => Console.WriteLine($"OnTransitioned = {t.Source}->{t.Destination}"));

    sm.Configure(State.A)
       .Permit(Trigger.X, State.B);

    sm.Configure(State.B)
        .InitialTransition(State.C);

    sm.Configure(State.C)
        .SubstateOf(State.B);

    sm.Fire(Trigger.X);
    
    Console.WriteLine($"Final state = {sm.State}");
}

public enum State { A, B, C }

public enum Trigger { X }
```

Output: 
```
OnTransitioned = A->B
Final state = C
```

Is this intended behavior? In my opinion, initial transition to state C should also call OnTransitioned handler and the output should be:
```
OnTransitioned = A->B
OnTransitioned = B->C
Final state = C
```

Using Stateless 5.1.5 in .NET Core 3.1

**Comment by HenningNT** [comment]
I think it should be...
```
OnTransitioned = A->C
Final state = C
```

This is a bug.

**Comment by DeepakParamkusam** [comment]
OnTransitionCompleted event is not available on v5.1.5 (latest nuget). However, I pulled the dev branch and ran the same test using OnTransitionCompleted event and the output is same:
```
OnTransitionCompleted = A->B
Final state = C
```

**Comment by HenningNT** [comment]
I see, looks like I have a bit more to do :-)

**Comment by HenningNT** [comment]
Fixed in 5.1.7
